lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -121,45 +121,18 @@ func (r *forwardInterceptor) resolveFromClient(
 		})
 
 	case ResolveHoldForwardAction_FAIL:
-		// Fail with an encrypted reason.
-		if in.FailureMessage != nil {
-			if in.FailureCode != 0 {
-				return status.Errorf(
-					codes.InvalidArgument,
-					"failure message and failure code "+
-						"are mutually exclusive",
-				)
-			}
-
-			// Verify that the size is equal to the fixed failure
-			// message size + hmac + two uint16 lengths. See BOLT
-			// #4.
-			if len(in.FailureMessage) !=
-				lnwire.FailureMessageLength+sha256.Size+2+2 {
-
-				return status.Errorf(
-					codes.InvalidArgument,
-					"failure message length invalid",
-				)
-			}
-
-			return r.htlcSwitch.Resolve(&htlcswitch.FwdResolution{
-				Key:            circuitKey,
-				Action:         htlcswitch.FwdActionFail,
-				FailureMessage: in.FailureMessage,
-			})
+		// Instantiate the fwdRes with base fields set.
+		fwdRes := &htlcswitch.FwdResolution{
+			Key:    circuitKey,
+			Action: htlcswitch.FwdActionFail,
 		}
 
-		code, err := unmarshalFailCode(in.FailureCode)
+		err := unmarshallFailureResolution(in, fwdRes)
```

Comment:
Don't think it's blocking, but we could have this instead return the new value and assign it here, s.t the new function is a pure function. Then tests can just exercise that the right value is returned vs asserting that the input is mutated as expected.

Reply:
I've been trying out multiple ways to make it a pure function, but after the changes in the later commit that extend this function it never really worked out in a clean way. So settled for this.